src/codelabs/iterables.md

```diff
@@ -0,0 +1,615 @@
+of the [Dart syntax](/samples).
+
+This codelab covers the following material:
+
+* How to read elements of an Iterable.
+* How to check if the elements of an Iterable match a condition.
+* How to filter the contents of an Iterable.
+* How to map the contents of an Iterable to a different value.
+
+Estimated time to complete this codelab: 60 minutes.
+
+{{site.alert.note}}
+  This page uses embedded DartPads to display examples and exercises.
+  {% include dartpads-embedded-troubleshooting.md %}
+{{site.alert.end}}
+
+## Why do you need collections
+
+Iterables are a kind of collection. Collections are classes that represented 
+a group of objects, also called "elements".
+
+A collection can be empty, or contain many elements in it. Depending on its purpose,
+a collection will follow a different data structure and have different implementations.
+These are some of the most common collection types:
+
+* [List][list class]: Used to read elements by their index.
+* [Set][set class]: Used to contain elements that can only occur once.
+* [Map][map class]: Used to read elements using a key.
+
+In Dart, both `List` and `Set` are `Iterable`, meaning that they share the same methods as the 
+`Iterable` class. 
+
+`Map` uses a different data structure internally depending on their implementation, 
+for example [HashMap][hashmap class] uses a hash-table,
+in which the elements (also called values) are obtained using a key. Elements of a `Map` can
+also be read as `Iterable`, with the properties `entries` or `values`.
+
+## What is an Iterable?
+
+An `Iterable` is a collection of elements that can be accessed sequentially.
+
+`Iterable`, in Dart, is an abstract class, meaning that you can't instantiate it directly.
+However you can create a new `Iterable` by creating a new `List` or a `Set`.
+ 
+This example shows a `List` of `int`, which is also an `Iterable` of `int`:
+
+{% prettify dart %}
+    Iterable<int> iterable = [1, 2, 3];
+{% endprettify %}
+
+The difference with a `List`, is that with the `Iterable` you can't 
+guarantee that reading elements by index will be efficient.
+`Iterable`, as opposed as `List`, does not have the `[]` operator.
+
+For example **the following code will give you an error**:
+
+{:.bad}
+{% prettify dart %}
+    Iterable<int> iterable = [1, 2, 3];
+    var value = iterable[![1]!];
+{% endprettify %}
+
+If you read elements with `[]`, the compiler will tell you that
+the operator `'[]'` isn't defined for the class `Iterable` which
+means that you can't use `[index]` in this case.
+
+You can instead read elements with `elementAt()`, which will step through
+the elements of the iterable until reaching that position.
+
+{% prettify dart %}
+    Iterable<int> iterable = [1, 2, 3];
+    var value = iterable.elementAt(1);
+{% endprettify %}
+
+Continue to the next section to learn more about how to access elements of an `Iterable`.
+
+## Reading elements
+
+The elements of an `Iterable` can be read sequentially.
```

Comment:
There's no need to mention Iterator before the sample. How about just changing these two paragraphs to this:

You can read the elements of an iterable sequentially, using a `for-in` loop.